### PR TITLE
[ci] Use numSplits instead of copy pasting test steps

### PR DESCRIPTION
**Pull request by daniel-goldstein**
The diff is horrendous but I just deleted the copy-pasted steps 1-n and changed the zero'th to be the new step.

Also fixed a bug in the brand new vectorized RunImage where the job name wasn't getting the index suffix.

### CODE REVIEW

File: build.yaml [hunk 1]
```diff
@@ -1007,143 +1008,7 @@ steps:
...
-      - from: /resources.tar.gz
-        to: /io/resources.tar.gz
-      - from: /hail-debug-test.jar
-        to: /io/hail-test.jar
-      - from: /splits.tar.gz
-        to: /io/splits.tar.gz
-    outputs:
-      - from: /io/test-output
-        to: /test-output-4
+        to: /test-output
```
Comment (danking) [inline build.yaml:1011]: Hmm this seems incorrect. All the test outputs will race to overwrite one another. I don’t think we care about the outputs though. Just remove them? Or add a template variable to outputs?

**Review by danking — DISMISSED** [review]
Minor point. Looks great!

File: build.yaml (hunk 1 above)
Comment (daniel-goldstein) [reply build.yaml:1011]: It's not exactly explicit so happy to hear alternatives

Comment (danking) [reply build.yaml:1011]: Ah. Hmm. A bit confusing. I don't have a better alternative though.

**Review by danking — APPROVED** [review]

**Pull request merged**